woocommerce/woocommerce-ios
# Shipping Labels: remote address validation and errors display

Part of #2973

## Description
As part of Shipping Labels M2, we should validate the shipping label address remotely and manage the errors. In this PR, we start to validate the address calling the remote API, and we also show the errors returned. In particular, if an error on the `address` field will be returned, an inline error will be shown. If a generic validation error will be returned, a top banner will be shown.

## Changes
- Now the `ShippingLabelAddress` model conforms to `Copiable`.
- Added the Top Banner View for managing the showing the address validation errors in `ShippingLabelAddressFormViewController`.
- Added the Done button in the Shipping Label Address Form that fires the validation action.
- Managed the cell `fieldError` in `ShippingLabelAddressFormViewController` for managing the errors at the specific level (just the address field return a specific errors).
- Implemented the handling of the various address fields when they changes.
- `ShippingLabelAddressFormViewModel`: managed the sections with the `fieldError` that will be added or removed based on the address validation error.
- `ShippingLabelAddressFormViewModel`: implemented a first version of the `validateAddress` API request for validating the shipping address.
- `ShippingLabelAddressTopBannerFactory`: implemented this new Top Banner Factoru that is useful for returning the `TopBannerView` configured with the data we needs in the Shipping Label Address validation.
- `TopBannerView`: fixed a bug that was now correctly hiding the title, because it's embedded inside a `titleStackView`.
- Implemented new tests for `ShippingLabelAddressFormViewModel`.
- Exposes `ShippingLabelAddressValidationError` to Yosemite.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in the Create Shipping Label form.
5. The screen for validating an address should be presented with all the information.
6. Try to use a wrong address (consider that the Country field, if shown in the expanded way, like `United States`, it's wrong. It will be handled in another PR because we should use the short version `US`) and press done. -> You should see the top banner on top reporting a generic error.
7. Try to fix the address and press done. -> If correctly validated, the banner on top should disappear.
8. Try to use an address that is correct, but with only the field `address wrong`... like `Bla bla bla`. Press done. -> You should see an inline error.

The loading indicator during the API request and other small things will be handled in future PRs.


## Screenshots
| Top banner error             | In-line error |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 12 Pro - 2021-02-23 at 17 15 37](https://user-images.githubusercontent.com/495617/108887114-180a2280-760a-11eb-96f5-b186f0c48577.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-02-23 at 17 29 10](https://user-images.githubusercontent.com/495617/108887119-19d3e600-760a-11eb-9ec6-5ab56a3e2683.png)


Update release notes:
